scripts/util.py

```diff
@@ -51,8 +64,74 @@ def read_source_root_path():
 def read_proof_root_path():
     return read_path_from_stdin("the 'proofs' directory (usually '.')")
 
-def read_litani_path():
-    return read_path_from_stdin("the litani script")
+def is_litani_submodule_in_path(path):
+    is_in_path = False
+    try:
+        sys.path.insert(1, path)
+        importlib.import_module("litani")
+        is_in_path = True
+    except ModuleNotFoundError:
+        print(f"Litani submodule was not found at directory: {path})")
+    del sys.path[1]
+    return is_in_path
+
+def add_litani_submodule(path):
+    """Adds the litani submodule to the specified path"""
+    try:
+        subprocess.run(
+             ["git", "submodule", "add", "--quiet", "--name", "litani", "--",
+             "https://github.com/awslabs/aws-build-accumulator.git", "litani"],
+             check=True, cwd=path)
+    except subprocess.CalledProcessError:
+        print(f"Could not add Litani submodule at path: {path}")
+
+def is_litani_installed():
+    return shutil.which("litani") is not None
+
+def determine_litani_origin():
+    choice = None
+    while True:
+        print(LITANI_DIALOG, file=sys.stderr)
+        choice = read_from_stdin()
+        if choice in ("1", "2", "3"):
+            break
+        print("You have entered an invalid value. Please choose again.")
+    return choice
+
+def install_litani_brew_package():
+    """Runs `brew install litani`. Exits if a failure occurs"""
+    try:
+        proc = subprocess.run(
+            ["brew", "install", "litani"],
+            check=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
Can I suggest that we don't buffer stderr here? I think it's more reassuring for the user to see it, whether there's an error or not. I think that might even be true of stdout. No need to hide it. Below, just do `sys.exit(1)` without logging.

(Even if you don't want to do what I suggest above, there's no need to do `decode()` below. Just pass `text=True` to `subprocess.run`)

Reply:
In the case that a user has already installed Litani using brew, removing `stderr=subprocess.PIPE` will result into the following message popping us (written to stderr). The message:

```
Warning: litani 1.22.0 is already installed and up-to-date.
To reinstall 1.22.0, run:
  brew reinstall litani
```
We're fine with that, right? I personally am fine with this message popping up and with removing the `stderr` as you pointed out.

Reply:
I think this is fine, but also this method doesn't even get called if Litani was already on the $PATH, right?